### PR TITLE
Expose the init_app() method in flask_fas_openid

### PR DESCRIPTION
**Pull request by pypingou**
Move the code to customize the flask app in this method and ensure there
is a flask app to customize before doing so.

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

### CODE REVIEW

**Comment by pypingou** [comment]
This was found out in the pagure re-architecturing

File: flask_fas_openid.py [hunk 1]
```diff
@@ -89,7 +89,18 @@ def __init__(self, app=None):
         self.postlogin_func = None
         self.app = app
         if self.app is not None:
-            self._init_app(app)
+            self.init_app(app)
+
+    def init_app(self, app):
+        """ Constructor for the Flask application. """
+        self.app = app
```
Comment (bowlofeggs) [inline flask_fas_openid.py:96]: It looks like this happens on line 90 above, which I think means you can also drop the app argument from this method.

**Review by bowlofeggs — APPROVED** [review]

File: flask_fas_openid.py (hunk 1 above)
Comment (pypingou) [reply flask_fas_openid.py:96]: The ``__init__`` method does not require ``app`` to be specified (defaulting to ``None``) while ``init_app`` does require ``app`` to be specified, so if ``__init__`` is called without ``app``, which is the case when you manually call ``init_app``, then ``self.app`` needs to be set in ``init_app`` :)

**Comment by pypingou** [comment]
Thanks for the review! :)

**Pull request merged**